### PR TITLE
Remove hardcoded -openstack requirement in image name.

### PR DESCRIPTION
**Pull request by imain**
Change the way we handle the image filenames so that we aren't requiring
that '-openstack' be in the image name.  Also rename some variables so
things are hopefully more clear.

Note this is related to https://github.com/openshift/installer/pull/3236

This commit should go in first and then the installer change should go in.

### CODE REVIEW

**Comment by imain** [comment]
Installer change: https://github.com/openshift/installer/pull/3236

File: get-resource.sh [hunk 1]
```diff
@@ -17,24 +17,25 @@ fi
...
+    RHCOS_IMAGE_FILENAME_QCOW=${RHCOS_IMAGE_FILENAME_RAW/.[gx]z}
+    IMAGE_FILENAME_EXTENSION=${RHCOS_IMAGE_FILENAME_RAW/$RHCOS_IMAGE_FILENAME_QCOW}
     IMAGE_URL=$(dirname $RHCOS_IMAGE_URL_STRIPPED)
 else
     echo "Unexpected image format $RHCOS_IMAGE_URL"
     exit 1
 fi
-RHCOS_IMAGE_FILENAME_COMPRESSED=${RHCOS_IMAGE_FILENAME_OPENSTACK/-openstack/-compressed}
+
+RHCOS_IMAGE_FILENAME_CACHED=cached-${RHCOS_IMAGE_FILENAME_QCOW}
```
Comment (derekhiggins) [inline get-resource.sh]: How about something like 
`RHCOS_IMAGE_FILENAME_CACHED=${RHCOS_IMAGE_FILENAME_OPENSTACK/.qcow2/-compressed.qcow2}`
this would maintain the existing API and remove the need for having "openstack" in the name.

Comment (imain) [reply get-resource.sh]: Hmm, I think I did the renaming such that it still has the same API.  I tested it locally and it came up with the same result.  I'll be doing full testing today.

Maybe I'm missing something here though?  Is there a flaw in my logic?

Comment (derekhiggins) [reply get-resource.sh:28]: What I meant was the filename of the compressed qcow2 file has changed but looking at the example I gave it also changes so forget I said anything.

Comment (imain) [reply get-resource.sh:28]: Right, I see.. sadly this requires changes here and in the installer.. and I haven't gotten them to work nicely yet.. :/

**Comment by imain** [comment]
This works now in my testing.

File: get-resource.sh (hunk 1 above)
Comment (imain) [reply get-resource.sh:28]: Yes, I think a symlink with the original substitution is what will be required.  If the name has -openstack in it that is.  Inserting a -openstack before .qcow2 is just an insertion where the original expected a substitution so I don't think that will fix the backwards compatibility.

File: get-resource.sh [hunk 2]
```diff
@@ -44,17 +45,26 @@ else
...
-    ln -sf "$RHCOS_IMAGE_FILENAME_OPENSTACK/$RHCOS_IMAGE_FILENAME_COMPRESSED" $FFILENAME
-    ln -sf "$RHCOS_IMAGE_FILENAME_OPENSTACK/$RHCOS_IMAGE_FILENAME_COMPRESSED.md5sum" "$FFILENAME.md5sum"
+    mv $TMPDIR $RHCOS_IMAGE_FILENAME_QCOW
+    ln -sf "$RHCOS_IMAGE_FILENAME_QCOW/$RHCOS_IMAGE_FILENAME_CACHED" $FFILENAME
+    ln -sf "$RHCOS_IMAGE_FILENAME_QCOW/$RHCOS_IMAGE_FILENAME_CACHED.md5sum" "$FFILENAME.md5sum"
 else
     rm -rf $TMPDIR
 fi
 
+# For backwards compatibility, if the rhcos image name contains -openstack, we want to
```
Comment (imain) [inline get-resource.sh:62]: So this makes it work with the current version of the installer (without my patch).  Also nice that we can get this in first and then the installer patch.

File: get-resource.sh [hunk 3]
```diff
@@ -44,17 +45,27 @@ else
...
+# create a symlink to the original in the old format.  The old format had a substitution
+# which required the existence of -openstack in the name.
+if [[ ${RHCOS_IMAGE_FILENAME_QCOW} == *"-openstack"* ]] ; then
+    cd "/shared/html/images/${RHCOS_IMAGE_FILENAME_QCOW}"
+    RHCOS_IMAGE_FILENAME_COMPRESSED=${RHCOS_IMAGE_FILENAME_QCOW/-openstack/-compressed}
+    ln -sf "$RHCOS_IMAGE_FILENAME_CACHED" "$RHCOS_IMAGE_FILENAME_COMPRESSED"
+    ln -sf "$RHCOS_IMAGE_FILENAME_CACHED.md5sum" "$RHCOS_IMAGE_FILENAME_COMPRESSED.md5sum"
+fi
```
Comment (hardys) [inline get-resource.sh:70]: If we did this earlier in the script, we could check for the existence of the old filename, and if it exists just rename it instead of downloading the same file again?

Comment (imain) [reply get-resource.sh:70]: Well we certainly could.  What are the chances it would be the same file?  This is mostly going to be on an upgrade.  My only real hesitation here is that I think it would be hard to test and it's a corner case optimization that adds one time value with more logic.

Comment (imain) [reply get-resource.sh:70]: Actually it might not be so hard to test.  Let me know and I can implement it, I just like things to stay simple. :)

Comment (hardys) [reply get-resource.sh:70]: Currently there's no (automated) way to update the bootimage on upgrade but you're right that I expect it would change on a major version upgrade, so perhaps it is a corner case but it would avoid re-downloading the old image in the case where a user forgets to update the image prior to the upgrade :)

Personally I think it's worth doing as downloads from non-local mirrors can be really slow, but I'm not blocking on it :)

**Comment by derekhiggins** [comment]
This has been here for a long time and the associated installer patch has been closed, is it still needed?
https://github.com/openshift/installer/pull/3236

I've tested it just now and all the parts still seem to work with dev-scripts completing successfully (without the installer patch)

```
drwxr-xr-x. 2 root root       4096 Sep 15 10:53 ./rhcos-46.82.202008260918-0-openstack.x86_64.qcow2
-rw-r--r--. 1 root root 1013907456 Sep 15 10:52 ./rhcos-46.82.202008260918-0-openstack.x86_64.qcow2/cached-rhcos-46.82.202008260918-0-openstack.x86_64.qcow2
-rw-r--r--. 1 root root         33 Sep 15 10:53 ./rhcos-46.82.202008260918-0-openstack.x86_64.qcow2/cached-rhcos-46.82.202008260918-0-openstack.x86_64.qcow2.md5sum
lrwxrwxrwx. 1 root root         56 Sep 15 10:53 ./rhcos-46.82.202008260918-0-openstack.x86_64.qcow2/rhcos-46.82.202008260918-0-compressed.x86_64.qcow2 -> cached-rhcos-46.82.202008260918-0-openstack.x86_64.qcow2
lrwxrwxrwx. 1 root root         63 Sep 15 10:53 ./rhcos-46.82.202008260918-0-openstack.x86_64.qcow2/rhcos-46.82.202008260918-0-compressed.x86_64.qcow2.md5sum -> cached-rhcos-46.82.202008260918-0-openstack.x86_64.qcow2.md5sum
-rw-r--r--. 1 root root 2433155072 Sep 15 10:50 ./rhcos-46.82.202008260918-0-openstack.x86_64.qcow2/rhcos-46.82.202008260918-0-openstack.x86_64.qcow2
lrwxrwxrwx. 1 root root        106 Sep 15 10:53 ./rhcos-ootpa-latest.qcow2 -> rhcos-46.82.202008260918-0-openstack.x86_64.qcow2/cached-rhcos-46.82.202008260918-0-openstack.x86_64.qcow2
lrwxrwxrwx. 1 root root        113 Sep 15 10:53 ./rhcos-ootpa-latest.qcow2.md5sum -> rhcos-46.82.202008260918-0-openstack.x86_64.qcow2/cached-rhcos-46.82.202008260918-0-openstack.x86_64.qcow2.md5sum

```

**Comment by imain** [comment]
Yeah sorry this was before I went on leave and for some reason I thought it had merged.  Still a valid problem to solve if we want to continue to pursue it.  I'll look at shardy's suggestion and see about a new revision.

**Comment by imain** [comment]
Sorry this is down in priority atm.  I'll get to it when I have a few free cycles

**Comment by imain** [comment]
This is still low priority but I rebased it and tested.  Still seems good.  If we could get this in I'll do up another installer patch.

**Comment by imain** [comment]
OK, if possible I'd like to see this go in and I'll do the installer patch.  Thanks!
